changelog/pending/20231207--ci--improve-file-diffing-in-ci-failures.yaml

```diff
@@ -0,0 +1,4 @@
+changes:
```

Comment:
Nit: This likely doesn't warrant a changelog entry as it's not really user facing.